Name: Fix duplicator stat limit check
Author: Adrian Siekierka <kontakt@asie.pl>

--- src.orig/SRC/ELEMENTS.PAS	2020-03-16 14:53:28.665824692 +0100
+++ src/SRC/ELEMENTS.PAS	2020-03-16 22:54:28.251467675 +0100
@@ -980,7 +980,7 @@ procedure ElementDuplicatorTick(statId:
 					if Board.Tiles[X - StepX][Y - StepY].Element = E_EMPTY then begin
 						sourceStatId := GetStatIdAt(X + StepX, Y + StepY);
 						if sourceStatId > 0 then begin
-							if Board.StatCount < 174 then begin
+							if Board.StatCount < MAX_STAT then begin
 								AddStat(X - StepX, Y - StepY,
 									Board.Tiles[X + StepX][Y + StepY].Element,
 									Board.Tiles[X + StepX][Y + StepY].Color,
